Mathlib/Data/Sum/Basic.lean

```diff
@@ -59,6 +59,20 @@ theorem «exists» {p : Sum α β → Prop} : (∃ x, p x) ↔ (∃ a, p (inl a)
     | Or.inr ⟨b, h⟩ => ⟨inr b, h⟩⟩
 #align sum.exists Sum.exists
 
+theorem forall_sum_pi (p : α ⊕ β → Sort*)
+    (q : (∀ a, p a) → Prop) :
+    (∀ a, q a) ↔ (∀ a b, q (Sum.rec a b)) :=
```

Comment:
Can you add the types of `a` and `b` here for clarity?

Reply:
done. I also made some arguments implict and improved some other names